pypsa/plot.py

```diff
@@ -139,7 +151,30 @@ def plot(network, margin=0.05, ax=None, geomap=True, projection=None,
                        "use `geomap` instead.")
         geomap = basemap
 
-    if cartopy_present and geomap:
+    if geomap:
+        if not (cartopy_present or basemap_present):
+            # Not suggesting Basemap since it is being deprecated
+            logger.warning("Cartopy needs to be installed to use `geomap=True`.")
+            geomap = False
+
+        # Use cartopy by default, fall back on basemap
+        use_basemap = False
+        use_cartopy = cartopy_present
+        if not use_cartopy:
+            use_basemap = basemap_present
```

Comment:
What's the argument behind duplicating the logic of the above 4 lines also on the top-level? Why not remove the top-level switch in lines 63-67, as well as `global` parameter?

Reply:
Or instead just use the global one.

Reply:
They are global because `use_basemap` and `use_cartopy` are modified from within `plot`, and used inside `draw_map`.
I included the logic in the top level to allow a user to call `draw_map` on its own.

In hindsight, it might be better to change the `if use_cartopy:` condition in `draw_map` to `if basemap_parameters=None:`. In fact, [using `{}` as default argument was a bad idea](https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments). I will update my PR!

Reply:
Fixed in my [latest commit](https://github.com/PyPSA/PyPSA/pull/58/commits/1088fc6247b52951983ab6ee94e0fd7d2a291af7).